core/src/main/java/co/aikar/commands/BaseCommand.java

```diff
@@ -64,69 +65,228 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+/**
+ * BaseCommand stores data about a single command tree.
+ * This includes {@link CommandManager who} manages it, what {@link RegisteredCommand subcommands} it has and so on.
+ */
 @SuppressWarnings("unused")
 public abstract class BaseCommand {
-
+    /**
+     * This is a field which contains the key in the {@link #subCommands} map for the method to catch any unknown
+     * argument to command states.
+     */
     public static final String CATCHUNKNOWN = "__catchunknown";
+
+    /**
+     * This is a field which contains the key in the {@link #subCommands} map for the method which is default for the
+     * entire (sub)command.
+     */
     public static final String DEFAULT = "__default";
+
+    /**
+     * Stores all the subcommands of this command.
+     */
     final SetMultimap<String, RegisteredCommand> subCommands = HashMultimap.create();
-    final Map<Class<?>, String> contextFlags = Maps.newHashMap();
+
+    final Map<Class<?>, String> contextFlags = Maps.newHashMap(); // TODO: Unknown use-case
+
+    /**
+     * Contains the manager which owns and handles this command.
+     */
+    CommandManager<?, ?, ?, ?, ?, ?> manager = null;
+
+    /**
+     * The command which owns this. This may be null if there are no owners.
+     */
+    @Nullable BaseCommand parentCommand = null;
+
+    /**
+     * A map of all registered commands under this.
+     * A registered command is any subcommand.
```

Comment:
lets drop the subcommand wording.

A registered command is not a subcommand. It is simply a command.

Subcommand is simply one access pattern to reach the command. But you can also use an alias and access it from a root command.